docs/api/apiv3/components/schemas/work_packages_by_project_model.yml

```diff
@@ -1,6 +1,6 @@
 # Schema: Work_Packages_by_ProjectModel
 ---
-type: object
+$ref: "./work_packages_model.yml"
```

Comment:
The response of both endpoints is created by the same service, just the scope differs. Hence, I suggest the following approach:

- remove schema `work_packages_by_project`
- reference `work_packages_model` directly from `project_work_packages.yml`

Reply:
That's fine with me. That's actually what I had in mind first, but I then thought that maybe you'd insist on having the different examples, as the `self` link differs:

`href: "/api/v3/work_packages"` vs. `href: "/api/v3/projects/14/work_packages"`

However, the rest is just repeated verbatim...

So yes, will remove the redundant models in these cases alltogether. Note that the `self` link then however will probably be not 100% as returned by OpenProject (not tested by myself so far, though).

The PR https://github.com/opf/openproject/pull/10178 goes exactly into this direction, so it's probably (mostly) fine?

We also seem to have the same massive repetition for `available_projects_for_XXX_model.yml`, and e.g.

At first I also thought that `available_assignees_model.yml`, `available_responsibles_model.yml`, `available_watchers_model.yml` are all just `users_model.yml`s, but this does not seem to be entirely true - while the schemas are not yet properly modelled, the examples differ in the `update` or `showUser` methods. Probably intentional and thus cannot be merged with `users_model.yml`?

Reply:
Point about the self link is true, but it shouldn't be a different schema though. Just the example differs. In the schema, both have a `self` link, which has only an `href`.

Reply:
#10178 does the same for attachments, I didn't check those yet in production, but will do now.

Reply:
Please have a look at the recent commit.